Sources/SwiftParser/Specifiers.swift

```diff
@@ -797,20 +798,53 @@ extension Parser {
         unexpected.append(misspelledThrows)
         if effectSpecifiers?.throwsClause == nil {
           if matchedSubset == S.CorrectThrowsTokenKinds.self {
-            synthesizedThrows = missingToken(spec)
+            let synthesizedThrows = missingToken(spec)
+            var lookahead = self.lookahead()
+            guard lookahead.consume(if: .leftParen) != nil,
+              lookahead.canParseType()
+            else {
+              synthesizedThrowsClause = RawThrowsClauseSyntax(
+                throwsSpecifier: synthesizedThrows,
+                leftParen: nil,
+                type: nil,
+                rightParen: nil,
+                arena: self.arena
+              )
+              break
+            }
+
+            let leftParen = self.consume(if: .leftParen)!
+            let type = self.parseType()
+            let rightParen = self.consume(if: .rightParen)
+
+            synthesizedThrowsClause = RawThrowsClauseSyntax(
+              throwsSpecifier: synthesizedThrows,
+              leftParen: missingToken(.leftParen),
+              type: self.withAllTokensMarkedMissing(source: type),
+              rightParen: rightParen != nil ? missingToken(.rightParen) : nil,
+              arena: self.arena
+            )
+
+            unexpected.append(leftParen)
+            unexpected += type.allTokens
```

Comment:
You should be able to shove the entire `type` into `unexpected` if you make it a `[RawSyntax]` instead of `[RawTokenSyntax]`. You can also remove the `allTokens` property then.